src/main/java/org/folio/processing/matching/loader/query/QueryHolder.java

```diff
@@ -42,11 +56,87 @@ public QueryHolder applyQualifierComparisonPart(Qualifier qualifier) {
     return this;
   }
 
+  public QueryHolder replaceFieldReference(String fieldPath, boolean isJson) {
+    return this
+      .replaceSqlFieldReference(fieldPath, isJson)
+      .replaceCqlFieldReference(fieldPath, isJson);
+  }
+
+  public QueryHolder replaceSqlFieldReference(String fieldPath, boolean isJson) {
+    sqlQuery = isJson ? updateSqlJson(fieldPath) : updateSqlNonJson(fieldPath);
+    return this;
+  }
+
+  public QueryHolder replaceCqlFieldReference(String fieldPath, boolean isJson) {
+    cqlQuery = isJson ? cqlQuery.replace(FIELD_NAME, fieldPath) : EMPTY;
+    return this;
+  }
+
   public String getSqlQuery() {
     return sqlQuery;
   }
 
   public String getCqlQuery() {
     return cqlQuery;
   }
+
+  private String updateSqlJson(String fieldPath) {
+    String fieldReference;
+    String arrayJoin = EMPTY;
+    // TODO provide support for searching in nested arrays
+    if (shouldJoinArray(fieldPath)) {
+      arrayJoin = prependArrayJoin(TABLE_NAME, fieldPath);
+      fieldReference = "field" + getFieldReference(substringAfter(fieldPath, ARRAY_SIGN));
+    } else {
+      fieldReference = join(TABLE_NAME, ".jsonb", getFieldReference(fieldPath));
+    }
+    return arrayJoin + sqlQuery.replace(FIELD_NAME, fieldReference);
+  }
+
+  private String updateSqlNonJson(String fieldName) {
```

Comment:
I think it will be better to rename this method(and similar ones), because it can causes misunderstanding